lldb/include/lldb/Target/ThreadPlanStack.h

```diff
@@ -165,6 +167,7 @@ class ThreadPlanStackMap {
   }
 
   ThreadPlanStack *Find(lldb::tid_t tid) {
+    std::lock_guard<std::recursive_mutex> guard(m_stack_map_mutex);
```

Comment:
Instead of having a mutex everywhere the `  using PlansList = std::unordered_map<lldb::tid_t, ThreadPlanStack *>;` is used, we could consider using an lldb `ThreadSafeDenseMap` instead.

Reply:
https://github.com/apple/llvm-project/blob/next/lldb/include/lldb/Core/ThreadSafeDenseMap.h

Reply:
TIL about `ThreadSafeDenseMap`. I don't think it would work: this class has non-general purpose operations that I think we'd expect the atomic locking over their duration. If we were to use `ThreadSafeDenseMap`, it seems these same operations would end up being smaller sequences of lock/unlock operations, breaking the atomicity we get from this implementation.